File: Java/employeeObjectArray.java
class employee {
    String employeeName = "";
    int employeeId = 0;
    employee(String eN, int eID) {
        this.employeeName = eN;
        this.employeeId = eID;
    }
    
public void outPutEmployeeInfo() {
    System.out.println("empoyee name is " + this.employeeName + " and this employee id is " + this.employeeId + "\n");
    }
}

class Main {
    public static void shuffleInOrder(employee[] sa) {
        for(int i = 0; i < sa.length; i++) {
            sa[i].outPutEmployeeInfo();
        }
    }
    
    public static void main(String[] args) {
        employee e1 = new employee("jason", 1875);
        employee e2 = new employee("karl", 8905);
        employee e3 = new employee("sara", 3459);
        employee e4 = new employee("jessica", 7621);
        
        employee[] eArr = {e1,e2,e3,e4};
        
        shuffleInOrder(eArr);
        
    }
}

File: C++/employeeObjectArray.cpp
#include <iostream>
#include <string>
#include <vector>

class Employee {
    std::string employeeName;
    int employeeId;

public:
    Employee(std::string eN, int eID) : employeeName(eN), employeeId(eID) {}

    void outPutEmployeeInfo() {
        std::cout << "employee name is " << employeeName << " and this employee id is " << employeeId << "\n";
    }
};

void shuffleInOrder(std::vector<Employee>& sa) {
    for (size_t i = 0; i < sa.size(); i++) {
        sa[i].outPutEmployeeInfo();
    }
}

int main() {
    Employee e1("jason", 1875);
    Employee e2("karl", 8905);
    Employee e3("sara", 3459);
    Employee e4("jessica", 7621);

    std::vector<Employee> eArr = {e1, e2, e3, e4};

    shuffleInOrder(eArr);

    return 0;
}
